provider/cmd/app/app.go

```diff
@@ -322,6 +349,11 @@ func Run(ctx context.Context, opts Options) error {
 		return runGRPCServer(ctx, setupLog, log, srv, opts)
 	})
 
+	g.Go(func() error {
+		setupLog.Info("Starting streaming server")
+		return runStreamingServer(ctx, setupLog, log, srv, opts)
```

Comment:
is error from runStreamingServer logged somewhere?

Reply:
I don't think. Did you read errgroup docs? because g.Wait() will return first error and you return errors on 2 places into errgroup. Did you tested it?

```text
Wait blocks until all function calls from the Go method have returned, then returns the first non-nil error (if any) from them.
```

Reply:
did you read code from lukas frank, where he logs each error for other components of code as machine events separately? So my suggestion follow standard pattern in libvirt-provider code. You cannot hide errors from other components because it can end in graceful shutdown process with really serious error and we won't know it because you will always ignore it.

Reply:
i will create issue for improve error logging.